### PR TITLE
[IMP] mail: use ResizablePanel in discuss

### PR DESCRIPTION
**Pull request by zel-odoo**
This commit introduces the ResizablePanel for ActionPanel in Discuss.
This allows the user to resize the ActionPanel in Discuss.

Peek 2024-07-19 15-24

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by phenix-factory** [comment]
Very cool! But it doesn't make sense on window sizes that are not desktop : 
image

File: addons/mail/static/src/discuss/core/common/action_panel.xml [hunk 1]
```diff
@@ -2,15 +2,22 @@
...
-            <t t-slot="default"/>
+        <ResizablePanel t-if="props.resizable and !env.inChatWindow" class="classNames" handleSide="'start'">
+            <t t-call="mail.ActionPanelContent"/>
+        </ResizablePanel>
+        <div t-else="" t-att-class="classNames">
+            <t t-call="mail.ActionPanelContent"/>
         </div>
     </t>
 
+    <t t-name="mail.ActionPanelContent">
```
Comment (alexkuhn) [inline addons/mail/static/src/discuss/core/common/action_panel.xml:13]: `mail.ActionPanelContent` => `mail.ActionPanel.content`. Follow BEM naming convention so seeing the template name gives clear traceability of component name and part name)

**Review by alexkuhn — APPROVED** [review]
LGTM otherwise

@robodoo delegate+

File: addons/mail/static/src/discuss/core/common/action_panel.scss [hunk 2]
```diff
@@ -1,3 +1,7 @@
+.o-mail-ActionPanel {
+    max-width: 45vw;
+}
+
```
Comment (phenix-factory) [inline addons/mail/static/src/discuss/core/common/action_panel.scss:4]: This rule doesn't work for the chatter : 
image

**Comment by zel-odoo** [comment]
I think 45vw rule is not ideal but the problematic behavior is there already in the master. I guess the key point is when the screen size is narrowed or strange screen ratio, we should not use the side panel. It should turn to use the full-screen panel as the chat window or in smartphone but we lack the ability to detect the screen of iPad sized ratio?

For the chatter, I will make the change that disables the resizable panel there.

**Comment by zel-odoo** [comment]
@robodoo r+

**Pull request closed without merging**